core/src/identity/secp256k1.rs

```diff
@@ -74,28 +68,30 @@ impl From<Keypair> for SecretKey {
 
 /// A Secp256k1 secret key.
 #[derive(Clone)]
-pub struct SecretKey(secp::key::SecretKey);
-
-/// View the bytes of the secret key.
-impl AsRef<[u8]> for SecretKey {
-    fn as_ref(&self) -> &[u8] {
-        &self.0[..]
-    }
-}
+pub struct SecretKey(secp256k1::SecretKey);
 
 impl SecretKey {
     /// Generate a new Secp256k1 secret key.
     pub fn generate() -> SecretKey {
-        SecretKey(secp::key::SecretKey::new(&mut secp::rand::thread_rng()))
+        let mut r = rand::thread_rng();
+        let mut b = [0; secp256k1::util::SECRET_KEY_SIZE];
+        // This is how it is done in `secp256k1::SecretKey::random` which
+        // we do not use here because it uses `rand::Rng` from rand-0.4.
```

Comment:
I had a stab at upgrading to rand 0.6 but the tests use [secp256k1-test](https://crates.io/crates/secp256k1-test) to compare results to the C lib and that crate is stuck on `rand` 0.4.